.github/workflows/javascript-format.yml

```diff
@@ -8,26 +8,48 @@ jobs:
             # Step 1: Checkout the repository
             - name: Checkout Code
               uses: actions/checkout@v4
+              with:
+                  token: ${{ secrets.GITHUB_TOKEN }}
+                  ref: ${{ github.head_ref || github.ref }}
 
             # Step 2: Set up Node.js environment
             - name: Set up Node.js
               uses: actions/setup-node@v4
               with:
-                  node-version: "18" # Specify your Node.js version
+                  node-version: "18"
 
-            # Step 3: Install Prettier and ESLint globally
+            # Step 3: Install Prettier
             - name: Install Prettier
-              run: |
-                  npm install -g prettier
+              run: npm install -g prettier
 
             # Step 4: Run Prettier to format code
-            - name: Run prettier
+            - name: Format JavaScript files
               run: |
+                  echo "Auto-formatting JavaScript files..."
                   prettier "**/*.js" --write
-                  git diff
-                  git reset --hard
-                  prettier --check "**/*.js"
+                  echo "Formatting complete"
+
+            # Step 5: Check for changes and commit if needed
+            - name: Commit changes
+              run: |
+                  # Get the original commit author info
+                  AUTHOR_NAME=$(git log -1 --pretty=format:'%an')
+                  AUTHOR_EMAIL=$(git log -1 --pretty=format:'%ae')
+
+                  # Use original author for the formatting commit
+                  git config --local user.email "$AUTHOR_EMAIL"
+                  git config --local user.name "$AUTHOR_NAME"
 
-            # Step 5: Report status
-            - name: Complete
-              run: echo "Formatting completed successfully!"
+                  # Check if there are any changes
+                  if [[ -n $(git status -s) ]]; then
+                      echo "Formatting changes detected, creating commit..."
+                      echo "Committing as: $AUTHOR_NAME <$AUTHOR_EMAIL>"
+                      git add -A
+                      git commit -m "[TASK] Auto-format JavaScript files with Prettier"
```

Comment:
can we make the commit message consistent with the style guide: https://github.com/ORNL/DataFed/pull/1501/files

I.e. maybe refactor, feat, or whatever is appropriate in there?